fix: Modernize Android configuration and improve documentation for solana_mobile_client

Immadominion opened the pull request:
This PR brings the solana_mobile_client package up to date with modern Flutter and Android best practices, and greatly improves the documentation for new and existing users, ensuring that it actually works. 

**Key changes:**

### Android Modernization
- Migrated all Android Gradle files to support the latest AGP 8.1.2 and Gradle 8.5, ensuring compatibility with the latest Flutter and Android Studio versions.
- Updated `compileSdkVersion` and `targetSdkVersion` to 34, and set the required `namespace` property in all relevant `build.gradle` files.
- Removed deprecated `package` attributes from all `AndroidManifest.xml` files, as required by AGP 8+.
- Fixed directory and package structure mismatches for `MainActivity` and ensured all package names, namespaces, and application IDs are consistent across Gradle files, manifests, and source code.
- Cleaned up and clarified manifest and Gradle settings to prevent build errors and improve maintainability.
- Addressed CMake/native build misconfigurations by removing unnecessary or incorrect native build references.

### Documentation Improvements
- Completely rewrote the `README.md` to provide a clear, concise, and attractive landing page for pub.dev.
- Added side-by-side screenshots to visually demonstrate the plugin in action.
- Included a feature list, usage instructions, platform support, and contribution guidelines.
- Linked to the official Solana Mobile Stack documentation for further reference.
- Ensured all referenced screenshots are included in the repository.


### Screenshots
![solana_mobile_client_example_ss1](https://github.com/user-attachments/assets/48e94208-1791-4409-b277-9f701725ab2d)
![solana_mobile_client_example_ss2](https://github.com/user-attachments/assets/a5dbe04e-880e-41b7-9d91-eee8f9984fe0)


### General
- Ensured all changes are limited to the solana_mobile_client package and do not affect unrelated packages in the monorepo.

---

**Motivation:**  
These changes make the package easier to use, more robust for modern Android/Flutter development, and more welcoming for new contributors and Solana Mobile developers. The improved documentation and screenshots will help users quickly understand the value and usage of the package.

Let me know if you have any questions or would like further adjustments!

ookami-kb reviewed (commented):
Thanks for the PR. Just a couple of minor things, otherwise looks good.

Immadominion commented:
Thanks for the feedback, and super quick response!

I've addressed the review comments in this update:

- Updated the platform support section in the README to reflect that iOS is not currently supported, as suggested.
- Changed the Android namespace in `android/build.gradle` to `com.solana.solana_mobile_client` for consistency and correctness - idk how I skipped that, lol.

Additionally, I further refined the README usage section to provide a more practical, minimal, and copy-pasteable example for new users. The new usage guide avoids Bloc as in the full example, focuses on the core package API, and includes clear comments and a minimal StatefulWidget demo. I believe this will make it much easier for developers to get started and understand how to integrate the package into their own apps.

Let me know if there’s anything else you’d like me to adjust, or any other good first issues for Flutter mobile with Solana!

ookami-kb commented on packages/solana_mobile_client/android/build.gradle:
```diff
@@ -26,6 +26,7 @@ apply plugin: 'kotlin-android'
 
 android {
     compileSdkVersion 31
+    namespace "com.solana.solana_mobile_client_example"
```
`com.solana.solana_mobile_client`, not `com.solana.solana_mobile_client_example` ;)

Immadominion commented:
Bump @ookami-kb 👋,

Just a follow up on the pr.

Immadominion commented:
Hello @ookami-kb , 

Just checking in, the package has not been updated on pub dev. 🙇🏻
